fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/rfc_dimension/Domain.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.dimension.rfc_dimension;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionRow;
+import com.yahoo.bard.webservice.data.dimension.metadata.StorageStrategy;
+
+import org.joda.time.DateTime;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * The client to retrieve dimension rows (dimension values).
+ *
+ * Cardinality and LastUpdated date are optional contracts that may or may not be available depending on the
+ * underlying domain implementation.
+ */
+public interface Domain {
+
+    String getName();
+
+    DomainSchema getSchema();
+
+    /**
+     * Returns the {@link StorageStrategy} of the dimension.
+     *
+     * @see {@link Dimension#getStorageStrategy()}
+     *
+     * @return the storage strategy of the dimension.
+     */
+    StorageStrategy getStorageStrategy();
+
+    /**
+     * Get the cardinality of the dimension, if reportable.
+     *
+     * @see {@link Dimension#getCardinality()}
+     *
+     * @return cardinality, if measurable, otherwise Optional.empty()
+     */
+    Optional<Integer> getCardinality();
```

Comment:
Should be `OptionalInt`?

Reply:
Personally, I don't care either way. `Optional<Integer>` has more capabilities (it has `map`, and similar methods, where as `OptionalInt` is just a glorified `null` check), though `OptionalInt` does save on autoboxing. The document does say `OptionalInt` though.